source/docs/configuration.md

```diff
@@ -12,62 +19,72 @@ title: Configuring Embark
   "buildDir": "dist/",
   "config": "config/",
   "versions": {
-    "web3": "1.0.0-beta.34",
-    "solc": "0.4.17",
+    "web3": "1.0.0-beta",
+    "solc": "0.4.25",
     "ipfs-api": "17.2.4"
   },
   "plugins": {
   },
   "options": {
+    "solc": {
+      "optimize": true,
+      "optimize-runs": 200
+    }
   }
 }
 </code></pre>
 
+Let's look at the different options and learn what they do and mean.
+
 ### contracts
 
-List of directories in which embark should look for contracts. These typically are GLOB expressions (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`)
+This is a list of directories in which Embark should look for Smart Contract files. These typically are globbing patterns (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`).
+
+### app
 
-### pipeline
+Everything inside `app` configures the assets of our application, which could be anything from HTML, JavaScript to CSS and other assets.
 
-The mapping between the output files and the source. e.g `"js/dapp.js": ["app/dapp.js"]` will run `app/dapp.js` though the embark pipeline, and output the result into `dist/js/dapp.js`
+  - **js/dapp.js** - This is the JavaScript bundle that contains our application. Specifically, all files that are defined in this option (`app/dapp.js`).
+  - **index.html** - The entry point of our application (`app/index.html`)
+  - **images** - All image assets of our application that we can find in `app/images/`.
```

Comment:
I like the explanations you put here, but I feel we could go a bit more general. Something like "JS files are compiled using Webpack to create a bundle, all other files types are simply copied."
It's just that with your explanation, it looks like the above is the only way to go.
Also, it's important to explain that the left-hand side is the destination and the right-hand side is the starting files. Also, that we can use globs.
Sorry if I worded everything badly. I'm not sure how we should put it up.